OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -252,23 +252,23 @@ public virtual Barrel CheckFire(Actor self, IFacing facing, Target target)
 		protected virtual void FireBarrel(Actor self, IFacing facing, Target target, Barrel barrel)
 		{
 			Func<WPos> muzzlePosition = () => self.CenterPosition + MuzzleOffset(self, barrel);
-			var legacyFacing = MuzzleOrientation(self, barrel).Yaw.Angle / 4;
+			Func<int> legacyFacing = () => MuzzleOrientation(self, barrel).Yaw.Angle / 4;
```

Comment:
We need to be careful here, because this is changing semantics.  Consider the case where a slow moving projectile is fired from an actor, which then turns after it has fired.  It would be reasonable to expect `args.Facing` to not change in this case, because the facing of the projectile doesn't depend on that of the firing actor once it has been launched.

It might be clearer if we could instead expose this as a `CurrentMuzzleFacing` or similar to avoid confusion?